### PR TITLE
Wazo 2671 wazo confd cli

### PR DESCRIPTION
**Pull request by pc-m**
This PR should be merged after https://github.com/wazo-platform/xivo-tools/pull/42

### CODE REVIEW

**Comment by pc-m** [comment]
recheck

**Comment by pc-m** [comment]
recheck

File: wazo_confd_cli/commands/endpoint.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            kwargs['tenant_uuid'] = parsed_args.tenant
+
+        result = self.app.client.endpoints_sip.list(**kwargs)
+        if not result['items']:
+            return (), ()
+
+        raw_items = result['items']
+        if parsed_args.template:
+
+            def predicate(item):
```
Comment (sduthil) [inline wazo_confd_cli/commands/endpoint.py]:
```suggestion
            def has_template(item):
```

File: wazo_confd_cli/commands/endpoint.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                for parent in item['templates']:
+                    if parent['uuid'] == parsed_args.template:
+                        return True
+                return False
+
+            raw_items = [item for item in raw_items if predicate(item)]
```
Comment (sduthil) [inline wazo_confd_cli/commands/endpoint.py]:
```suggestion
            raw_items = [item for item in raw_items if has_template(item)]
```

**Review by sduthil — CHANGES_REQUESTED** [review]
I think a `--prefix` option would be nice too, in order to use `wazo-confd-cli` from another machine, e.g. with Docker:
```
docker run -it wazoplatform/wazo-confd-cli --hostname wazo.example.com --port 443 --prefix /api/confd --token mytoken user list
```

**Comment by sduthil** [comment]
Also, when trying the `wazo-confd-cli`, there is always an error message at the start (even though the CLI commands are working):

```
Could not read config dir /root/.config/wazo-confd-cli: [Errno 2] No such file or directory: '/root/.config/wazo-confd-cli'
```

File: wazo_confd_cli/commands/endpoint.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        headers = self.extract_column_headers(raw_items[0])
```
Comment (sduthil) [inline wazo_confd_cli/commands/endpoint.py:79]: There is no validation that `raw_items` is empty.
If I run `wazo-confd-cli ... endpoint sip list --template does-not-exist`, then I get a Traceback on this line.

**Comment by pc-m** [comment]
Thanks for the review, I fixed all reported problems. Some of them have been force pushed into the commit that adds the `--template` argument. I also changed the parsing of the https options because we could not specify the `https` and `verify_certificate` options that were in a mutually exclusive group

**Review by sduthil — APPROVED** [review]

**Pull request merged**